src/Core/Context/CurrentContext.cs

```diff
@@ -362,8 +362,9 @@ public async Task<bool> ManageResetPassword(Guid orgId)
 
         public async Task<bool> ManageBilling(Guid orgId)
         {
-            return await OrganizationOwner(orgId) &&
-                (await ProviderUserForOrgAsync(orgId) || ProviderIdForOrg(orgId) == null);
+            return ProviderIdForOrg(orgId) == null
```

Comment:
I feel like it's still a bit non-intuitive what ProviderIdForOrg does, how about extracting it to a variable?
```suggestion
			var managedByProvider = ProviderIdForOrg(orgId) == null;
            return managedByProvider
```

Reply:
Done, but flipped it for readability